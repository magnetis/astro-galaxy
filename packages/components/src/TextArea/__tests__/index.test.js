import React from 'react';
import { TextAreaWrapper, TextArea, TextAreaLabel } from '../index';

function mount(props = {}) {
  return rendererCreateWithTheme(
    <TextAreaWrapper>
      <TextArea {...props} />
      <TextAreaLabel>Label</TextAreaLabel>
    </TextAreaWrapper>
  );
}

describe('TextArea', () => {
  it('should render', () => {
    const wrapper = mount();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render disabled', () => {
    const wrapper = mount({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });
});
