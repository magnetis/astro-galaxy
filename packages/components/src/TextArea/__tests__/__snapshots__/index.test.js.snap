// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  padding: 25px 16px 5px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  background-color: #ffffff;
  border: 1px solid #597183;
  border-radius: 8px;
}

.c1:disabled {
  color: #afbec9;
  cursor: not-allowed;
  background-color: #f6f7f8;
  border: 1px solid #e9eef2;
}

.c1:focus {
  border: 1px solid #159ce4;
  outline: none;
}

.c1:focus + label {
  top: 8px;
  font-weight: 700;
  font-size: 12px;
  color: #159ce4;
}

.c2 {
  position: absolute;
  top: 16px;
  left: 16px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #8296a4;
  pointer-events: none;
  -webkit-transition: all 0.2s ease;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <textarea
    className="c1"
  />
  <label
    className="c2"
  >
    Label
  </label>
</div>
`;

exports[`TextArea should render disabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  padding: 25px 16px 5px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  background-color: #ffffff;
  border: 1px solid #597183;
  border-radius: 8px;
}

.c1:disabled {
  color: #afbec9;
  cursor: not-allowed;
  background-color: #f6f7f8;
  border: 1px solid #e9eef2;
}

.c1:focus {
  border: 1px solid #159ce4;
  outline: none;
}

.c1:focus + label {
  top: 8px;
  font-weight: 700;
  font-size: 12px;
  color: #159ce4;
}

.c2 {
  position: absolute;
  top: 16px;
  left: 16px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #8296a4;
  pointer-events: none;
  -webkit-transition: all 0.2s ease;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={true}
  />
  <label
    className="c2"
  >
    Label
  </label>
</div>
`;
